packages/axelar-wasm-std/src/voting.rs

```diff
@@ -130,21 +130,23 @@ impl fmt::Display for PollID {
 
 pub trait Poll {
     // errors if the poll is not finished
-    fn tally(&mut self, block_height: u64) -> Result<PollResult, Error>;
+    fn complete(&mut self, block_height: u64) -> Result<PollResult, Error>;
     // errors if sender is not a participant, if sender already voted, if the poll is finished or
     // if the number of votes doesn't match the poll size
     fn cast_vote(
         &mut self,
         block_height: u64,
         sender: &Addr,
         votes: Vec<bool>,
-    ) -> Result<PollStatus, Error>;
+    ) -> Result<(), Error>;
```

Comment:
I would prefer a signature like this:
`fn cast_vote(self, block_height: u64, sender: &Addr, votes: Vec<bool>) -> Result<Self, Error>`
Because the function owns self, it can still do mutation internally, but this way we get a cleaner immutable interface and don't have these awkward void returns